src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -13948,6 +13948,127 @@ def dt_strftime(self, date_format: str) -> None:
             "Snowpark pandas doesn't yet support the method 'Series.dt.strftime'"
         )
 
+    def topn(
+        self, n: int, columns: IndexLabel, keep: str, ascending: bool
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Return the top 'n' rows ordered by 'columns'..
+
+        Args:
+            n: Number of rows to return.
+            columns: Column label(s) to order by.
+            keep: {'first', 'last', 'all'}
+              Where there are duplicate values:
+              first : prioritize the first occurrence(s)
+              last : prioritize the last occurrence(s)
+              all : do not drop any duplicates, even it means selecting more than n items.
+            ascending: {True, False}
+              If True return smallest n values otherwise return largest n values.
+
+        Returns:
+            SnowflakeQueryCompiler
+        """
+        if keep not in ("first", "last", "all"):
+            raise ValueError('keep must be either "first", "last" or "all"')
+        if keep == "all":
+            method_name = "nsmallest" if ascending else "nlargest"
+            ErrorMessage.not_implemented(
+                f"Snowpark pandas method '{method_name}' doesn't yet support parameter keep='all'"
+            )
+        # Special case handling for unnamed series. 'columns' passed from frontend layer
+        # will be None, replace it with MODIN_UNNAMED_SERIES_LABEL.
+        if columns is None:
+            data_labels = self._modin_frame.data_column_pandas_labels
+            if len(data_labels) == 1 and data_labels[0] == MODIN_UNNAMED_SERIES_LABEL:
+                columns = MODIN_UNNAMED_SERIES_LABEL
+
+        if not isinstance(columns, list):
+            columns = [columns]
+
+        # Native pandas returns empty dataframe if n is negative. Set it to zero to
+        # provide same behavior.
+        n = max(0, n)
+        # Native pandas returns empty dataframe if 'columns' is a empty array. Set n
+        # to zero to provide same behavior.
+        if len(columns) == 0:
+            n = 0
+
+        # We need row position to break tie for duplicate values.
+        internal_frame = self._modin_frame.ensure_row_position_column()
```

Comment:
do you need the row position column? or just the ordering column? If we do not need the row position column, we can just use the current ordering columns

Reply:
good point. we can work with existing ordering columns.